bahmnicore-api/src/test/java/org/bahmni/module/bahmnicore/forms2/mapper/FormDetailsMapperTest.java

```diff
@@ -0,0 +1,238 @@
+        Whitebox.setInternalState(FormType.class, "FORM_BUILDER_FORMS", formType);
+        when(formType.get()).thenReturn("v2");
+
+        Collection<FormDetails> formDetailsCollection = FormDetailsMapper
+                .createFormDetails(singletonList(obs), formType);
+
+        assertEquals(1, formDetailsCollection.size());
+
+        FormDetails formDetails = formDetailsCollection.iterator().next();
+        assertEquals("v2", formDetails.getFormType());
+        assertEquals(formName, formDetails.getFormName());
+        assertEquals(formVersion, formDetails.getFormVersion());
+
+        verifyCommonData(formDetails);
+
+        verify(obs, times(2)).getFormFieldPath();
+        verifyStatic(VerificationModeFactory.times(1));
+        FormUtil.getFormNameFromFieldPath(formFieldPath);
+        verifyStatic(VerificationModeFactory.times(1));
+        FormUtil.getFormVersionFromFieldPath(formFieldPath);
+        verify(formType, times(1)).get();
+
+        verifyCommonMockCalls();
+
+    }
+
+    @Test
+    public void shouldReturnFormDetailsFromGivenObsAndFormTypeOfAllObservationTemplates() {
+
+        FormType formType = mock(FormType.class);
+        Whitebox.setInternalState(FormType.class, "ALL_OBSERVATION_TEMPLATE_FORMS", formType);
+        when(formType.get()).thenReturn("v1");
+
+        Concept concept = mock(Concept.class);
+        when(obs.getConcept()).thenReturn(concept);
+        ConceptName conceptName = mock(ConceptName.class);
+        when(concept.getName()).thenReturn(conceptName);
+        String obsName = "some obs name";
+        when(conceptName.getName()).thenReturn(obsName);
+
+        Collection<FormDetails> formDetailsCollection = FormDetailsMapper
+                .createFormDetails(singletonList(obs), formType);
+
+        assertEquals(1, formDetailsCollection.size());
+
+        FormDetails formDetails = formDetailsCollection.iterator().next();
+
+        assertEquals("v1", formDetails.getFormType());
+        assertEquals(formDetails.getFormName(), obsName);
+        assertEquals(0, formDetails.getFormVersion());
+        verifyCommonData(formDetails);
+
+        verify(formType, times(1)).get();
+        verifyCommonMockCalls();
+
+    }
+
+    @Test
+    public void shouldReturnFormDetailsWithTwoProvidersFromGivenTwoObsAndFormTypeOfFormBuilder() {
+
+        String anotherObsFormFieldPath = "FormName.2/2-0";
+        String anotherProviderName = "Another Super Man";
+        String anotherProviderUuid = "Another provider-uuid";
+
+        Obs anotherObs = mock(Obs.class);
+        User anotherCreator = mock(User.class);
+        PersonName anotherPersonName = mock(PersonName.class);
+
+        when(anotherObs.getEncounter()).thenReturn(encounter);
+        when(anotherObs.getCreator()).thenReturn(anotherCreator);
+        when(anotherObs.getFormFieldPath()).thenReturn(anotherObsFormFieldPath);
+        when(FormUtil.getFormNameFromFieldPath(anotherObsFormFieldPath)).thenReturn(formName);
```

Comment:
No need to mock FormUtil static methods. If the test data and code is written properly, there should be no need for mocking utils methods such as that. What are we benefitting from with such mocking?